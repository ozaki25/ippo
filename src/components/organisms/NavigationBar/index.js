import React from 'react';
import { Alignment, Button, Navbar } from '@blueprintjs/core';
import propTypes from 'prop-types';
import ROUTES from 'src/constants/routes';

const NavigationBar = ({ history, firebase, authUser }) => (
  <Navbar>
    <Navbar.Group align={Alignment.LEFT}>
      <Navbar.Heading>IPPO</Navbar.Heading>
      <Navbar.Divider />
    </Navbar.Group>
    <Navbar.Group align={Alignment.RIGHT}>
      <Button icon="wrench" minimal onClick={() => history.push(ROUTES.Publish)} />
      {authUser && <Button icon="log-out" minimal onClick={firebase.doSignOut} />}
    </Navbar.Group>
  </Navbar>
);

NavigationBar.displayName = 'NavigationBar';

NavigationBar.propTypes = {
  history: propTypes.shape({
    push: propTypes.func,
  }),
  firebase: propTypes.shape({
    doSignOut: propTypes.func,
  }),
  authUser: propTypes.object,
};

NavigationBar.defaultProps = {
  history: null,
  firebase: null,
  authUser: null,
};

export default NavigationBar;
