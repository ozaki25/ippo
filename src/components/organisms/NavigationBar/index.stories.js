import React from 'react';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import NavigationBar from '.';

export default {
  title: 'organisms/NavigationBar',
};

const props = {
  title: text('title', 'タイトル'),
  history: {
    push: action('push'),
    replace: action('replace'),
    goBack: action('back'),
  },
  firebase: {
    doSignOut: action('signout'),
  },
};

const authUser = {
  uid: 'abc',
  displayName: text('name', 'テストユーザ'),
};

export const ログイン未済 = () => <NavigationBar {...props} />;

export const ログイン済み = () => (
  <NavigationBar {...props} authUser={authUser} />
);

export const 戻るあり = () => <NavigationBar {...props} back />;

export const 全量 = () => <NavigationBar {...props} back authUser={authUser} />;
