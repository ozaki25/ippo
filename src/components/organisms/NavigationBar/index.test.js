import React from 'react';
import { snapshot } from 'test/helpers';
import NavigationBar from '.';

const props = {
  title: 'タイトル',
  history: {
    push: jest.fn(),
    replace: jest.fn(),
    goBack: jest.fn(),
  },
  firebase: {
    doSignOut: jest.fn(),
  },
};

const authUser = {
  uid: 'abc',
  displayName: 'テストユーザ',
};

snapshot('NavigationBar/nologin', <NavigationBar {...props} />);

snapshot('NavigationBar/login', <NavigationBar {...props} authUser={authUser} />);

snapshot('NavigationBar/back', <NavigationBar {...props} back />);

snapshot('NavigationBar/all', <NavigationBar {...props} back authUser={authUser} />);
