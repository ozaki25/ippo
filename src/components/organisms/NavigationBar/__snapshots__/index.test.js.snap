// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NavigationBar/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <NavigationBar
    authUser={null}
    firebase={null}
    history={Object {}}
  >
    <Blueprint3.Navbar>
      <div
        className="bp3-navbar"
      >
        <Blueprint3.NavbarGroup
          align="left"
        >
          <div
            className="bp3-navbar-group bp3-align-left"
          >
            <Blueprint3.NavbarHeading>
              <div
                className="bp3-navbar-heading"
              >
                IPPO
              </div>
            </Blueprint3.NavbarHeading>
            <Blueprint3.NavbarDivider>
              <div
                className="bp3-navbar-divider"
              />
            </Blueprint3.NavbarDivider>
          </div>
        </Blueprint3.NavbarGroup>
        <Blueprint3.NavbarGroup
          align="right"
        >
          <div
            className="bp3-navbar-group bp3-align-right"
          >
            <Blueprint3.Button
              icon="wrench"
              minimal={true}
              onClick={[Function]}
            >
              <button
                className="bp3-button bp3-minimal"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                <Blueprint3.Icon
                  icon="wrench"
                  key="leftIcon"
                >
                  <span
                    className="bp3-icon bp3-icon-wrench"
                    icon="wrench"
                  >
                    <svg
                      data-icon="wrench"
                      height={16}
                      viewBox="0 0 16 16"
                      width={16}
                    >
                      <desc>
                        wrench
                      </desc>
                      <path
                        d="M15.83 3.7l-3.06 3.05-2.84-.7-.7-2.83L12.29.17a5.004 5.004 0 0 0-4.83 1.29 4.967 4.967 0 0 0-1.12 5.36L.58 12.58c-.36.36-.58.86-.58 1.41 0 1.1.9 2 2 2 .55 0 1.05-.22 1.41-.59l5.77-5.77c1.79.69 3.91.33 5.35-1.12 1.32-1.3 1.74-3.15 1.3-4.81z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                <Blueprint3.Icon
                  key="rightIcon"
                />
              </button>
            </Blueprint3.Button>
          </div>
        </Blueprint3.NavbarGroup>
      </div>
    </Blueprint3.Navbar>
  </NavigationBar>
</MockRouter>
`;
